include/nlohmann/detail/input/input_adapters.hpp

```diff
@@ -45,6 +46,27 @@ struct input_adapter_protocol
 /// a type to simplify interfaces
 using input_adapter_t = std::shared_ptr<input_adapter_protocol>;
 
+/*!
+Input adapter for stdio file access. This adapter read only 1 byte and do not use any
+ buffer. This adapter is a very low level adapter. This adapter
+*/
+class file_input_adapter : public input_adapter_protocol
+{
+  public:
+    explicit file_input_adapter(const FILE* file)  noexcept
+        : file(file)
+    {}
+
+    std::char_traits<char>::int_type get_character() noexcept override
+    {
+        return fgetc(const_cast<FILE*>(file));
+    }
+  private:
+    /// the file pointer to read from
+    const FILE* file;
```

Comment:
Change to ```FILE * file;``` and remove the "const_cast" and the const from the constructor. (Using const_cast in this way is undefined behavior and there's no reason to take the FILE* by const anyway).

Reply:
updated as per your comment